### PR TITLE
Issue 1954 / Add if statement prior setting date

### LINKED ISSUE #1954: Event: Cancelled event changes to draft after clicking the scheduling calendar
## Description
In cancelled event when I click the 'Not visible or scheduled', the status 'cancelled' changes to 'draft'. 
## Steps to reproduce
1. Go to any cancelled event
2. Click on 'Not visible or scheduled'. Don't set any date. Just click outside of calendar box.

### Expected Behaviour
Event status should be remain cancelled.
### Actual Behaviour
Event status changes to draft
## Screenshots (if you have any)

https://github.com/zetkin/app.zetkin.org/assets/77925373/61012151-a59f-43e3-af33-ed9f7a63d391

### PR DESCRIPTION
**Pull request by a-jaxell**
## Description

This PR resolves a bug with regarding Event status when cancelling a drafted event.

Opening and closing the datepicker caused the date value being set to null. And then changing the status because of date value.

As far as my understanding goes, event status is dependent on date. So Im thinking this was working as intended from a point of view since an event without a date is considered a draft? And if the date gets set to the future it should be scheduled and so on... 

I have added an if statement that checks date before setting it so that the 'cancelled' status sticks around.

## Screenshots

https://github.com/zetkin/app.zetkin.org/assets/97404672/b8377b03-d8ec-4d51-ad34-e40e7f1a66ec

## Changes

* If statement validates date before setPublish in handleDateChange.

## Notes to reviewer

1. Go to / create any event
2. Click 'Cancel Event'
3. Click 'Not visible or scheduled'
4. Click outside date picker
5. Event status should not change

## Related issues
Resolves #1954

### CODE REVIEW

File: src/features/events/components/EventActionButtons.tsx [hunk 1]
```diff
@@ -43,7 +43,9 @@ const EventActionButtons: React.FunctionComponent<EventActionButtonsProps> = ({
     setPublished(null);
   };
   const handleChangeDate = (date: string | null) => {
-    setPublished(date);
+    if (date) {
```
Comment (richardolsson) [inline src/features/events/components/EventActionButtons.tsx:46]: I think this should be compared with the previous value. Because right now, I can still reproduce the bug if I publish the event first. That way, opening the date picker and clicking outside of it calls this function with the date that was already there, and that reproduces the bug.

Comment (a-jaxell) [reply src/features/events/components/EventActionButtons.tsx]: I have added a check against the previous date as you suggested.

However I have now noticed that if you set a date further ahead in the future it changes the date to scheduled and that 'Scheduled' remains as the event status regardless if you cancel or restore. 

Should future cancelled events show as scheduled? It seems a bit unintuitive..

Comment (richardolsson) [reply src/features/events/components/EventActionButtons.tsx]: Nice work trying different edge cases! It sounds to me like `getEventState()` needs to be improved. Ideally, there should be tests for it that test these cases. Do you want to check it out?

Comment (a-jaxell) [reply src/features/events/components/EventActionButtons.tsx]: Yes, I'll have a look! :)

**Review by richardolsson — APPROVED** [review]
I think this is working now! I agree that there may also be bugs with `getEventState()` but I think those should be filed and fixed separately.

If you want to try and describe the bug in a separate issue, @a-jaxell, that would be great!
